Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoASA.yaml

```diff
@@ -0,0 +1,180 @@
+Parser:
+  Title: Authentication ASIM parser for Cisco Device Logon Events
+  Version: '0.1.0'
+  LastUpdated: 6 Jul 2023
+Product:
+  Name: M365 Defender for EndPoint
+Normalization:
+  Schema: Authentication
+  Version: '0.1.0'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing authentication events, collected from Cisco ASA devices, to the ASIM Authentication schema.
+ParserName: ASimAuthenticationCiscoASA
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser = (
+    disabled:bool=false
+  ){
+    let DeviceEventClassIDLookup = datatable (DeviceEventClassID:string, EventResultDetails:string, EventType:string, EventResult:string, EventSubType:string, DvcAction:string)
+    [
+        "113004", "", "Logon", "Success", "Remote", "",
+        "113005", "Other", "Logon", "Failure", "Remote", "",
+        "113006", "Logon violates policy", "Logon", "Failure", "Remote", "Blocked",
+        "113008", "", "Logon", "Success", "Remote", "",
+        "113010", "", "Logon", "Success", "Remote", "",
+        "113012", "", "Logon", "Success", "Remote", "",
+        "113019", "", "Logoff", "Success", "", "",
+        "113039", "", "Logon", "Success", "Remote", "",
+        "315011", "", "Logoff", "Success", "", "",
+        "502103", "", "Elevate", "Success", "AssumeRole", "",
+        "605004", "Other", "Logon", "Failure", "Remote", "",
+        "605005", "", "Logon", "Success", "Remote", "",
+        "611101", "", "Logon", "Success", "Remote", "",
+        "611102", "Other", "Logon", "Failure", "Remote", "",
+        "611103", "", "Logoff", "Success", "", "",
+        "713198", "Logon violates policy", "Logon", "Failure", "Remote", "Blocked",
+        "716002", "", "Logoff", "Success", "", "",
+        "716038", "", "Logon", "Success", "Remote", "",
+        "716039", "Other", "Logon", "Failure", "Remote", "",
+        "716040", "Other", "Logon", "Failure", "Remote", "",
+        "722022", "", "Logon", "Success", "Remote", "",
+        "722023", "", "Logoff", "Success", "", "",
+        "722028", "", "Logoff", "Success", "", "",
+        "722037", "", "Logoff", "Success", "", ""
```

Comment:
I found additional potential authn logs, can you please have a look?
https://www.cisco.com/c/en/us/td/docs/security/asa/syslog/b_syslog/syslogs-sev-level.html
%ASA-2-772003: PASSWORD: session login failed, user username, IP ip, cause: password expired
%ASA-2-772006: REAUTH: user username failed authentication
%ASA-3-772002: PASSWORD: console login warning, user username, cause: password expired
%ASA-3-772004: PASSWORD: session login failed, user username, IP ip, cause: password expired
%ASA-2-772006: REAUTH: user username failed authentication

Also, I noticed that the number after ASA- is the severity. Can we try look again and add any authentication specific IDs having atleast 1,2 and 3 as Severity?

Reply:
Done

Reply:
Not yet